DFC.ServiceTaxonomy.PageLocation/Handlers/PageLocationPartHandler.cs

```diff
@@ -1,16 +1,89 @@
-﻿using System.Threading.Tasks;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 using DFC.ServiceTaxonomy.PageLocation.Models;
 using OrchardCore.ContentManagement;
 using OrchardCore.ContentManagement.Handlers;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.Title.Models;
+using YesSql;
 
 namespace DFC.ServiceTaxonomy.PageLocation.Handlers
 {
     public class PageLocationPartHandler : ContentPartHandler<PageLocationPart>
     {
+        private readonly ISession _session;
+
+        public PageLocationPartHandler(ISession session)
+        {
+            _session = session;
+        }
+
         public override Task InitializingAsync(InitializingContentContext context, PageLocationPart part)
         {
             part.Apply();
             return Task.CompletedTask;
         }
+
+        public override async Task CloningAsync(CloneContentContext context, PageLocationPart part)
+        {
+            string title = context.CloneContentItem.Content[nameof(TitlePart)][nameof(TitlePart.Title)];
+            string urlName = context.CloneContentItem.Content[nameof(PageLocationPart)][nameof(PageLocationPart.UrlName)];
+            string fullUrl = context.CloneContentItem.Content[nameof(PageLocationPart)][nameof(PageLocationPart.FullUrl)];
+
+            IEnumerable<ContentItem> pages = await _session.Query<ContentItem, ContentItemIndex>(x => x.ContentType == "Page" && x.Latest).ListAsync();
+
+            string urlSearchFragment = fullUrl.EndsWith("-clone") ?
```

Comment:
I think we should have an unit test to test the url & title generation.

Reply:
Done!